dbmongo/lib/engine/purgeBatch.go

```diff
@@ -1,114 +1,131 @@
+		log.Println(tempDBprefix+strconv.Itoa(i)+": ", query)
+		_, err := Db.DB.C("RawData").Find(query).MapReduce(&job, nil)
+		if err != nil {
+			fmt.Println(tempDBprefix+strconv.Itoa(i)+": error ", err.Error())
 		}
-		numQuery++
-
-		go MRroutine(
-			job,
-			chunkQuery,
-			tempDbName,
-			/*Origin collection name = */ "RawData",
-			/*Waitgroup = */ &w,
-			tempDBChannel,
-		)
 	}
+	wg.Done()
+}
 
-	// On consomme les objets dans tempDBChannel
-	go func() {
-		for tempDbName := range tempDBChannel {
-			tempDbNames = append(tempDbNames, tempDbName)
-		}
-	}()
+// PurgeBatch permet de supprimer tous les batch consécutifs au un batch donné dans RawData
+func PurgeBatch(batch base.AdminBatch) error {
+	chunks, err := ChunkCollection(viper.GetString("DB"), "RawData", viper.GetInt64("chunkByteSize"))
+	if err != nil {
+		return fmt.Errorf("chunkCollection a échoué: %s", err.Error())
+	}
+	queries := chunks.ToQueries(bson.M{}, "_id")
+	queryChan := queriesToChan(queries)
 
-	w.waitGroup.Wait()
-	close(tempDBChannel)
+	functions, err := loadJSFunctions("purgeBatch")
+	if err != nil {
+		return err
+	}
 
-	db, _ := mgo.Dial(viper.GetString("DB_DIAL"))
-	db.SetSocketTimeout(720000 * time.Second)
+	baseJob := mgo.MapReduce{
+		Map:      functions["map"].Code,
+		Reduce:   functions["reduce"].Code,
+		Finalize: functions["finalize"].Code,
+		Scope: bson.M{
+			"fromBatchKey": batch.ID.Key,
+			"f":            functions,
+		},
+	}
 
-	if clearTempCollections {
-		for _, tempDbName := range tempDbNames {
-			pipeline := []bson.M{
-				bson.M{
-					"$merge": bson.M{
-						"into": bson.M{
-							"coll":        "RawData",
-							"db":          viper.GetString("DB"),
-							"whenMatched": "merge",
-						},
-					},
-				},
-			}
-
-			pipe := db.DB(tempDbName).C("TemporaryCollection").Pipe(pipeline)
-			var result []interface{}
-			err = pipe.AllowDiskUse().All(&result)
-			if err != nil {
-				w.add("errors", 1, -1)
-			} else {
-				err = db.DB(tempDbName).DropDatabase()
-				if err != nil {
-					w.add("errors", 1, -1)
-				}
-			}
-		}
+	wg := sync.WaitGroup{}
+	var id = 0
+	var idMutex sync.Mutex
+	var tempDBs []string
+	for i := 0; i < viper.GetInt("MRthreads"); i++ {
+		wg.Add(1)
+		go MRChunks(queryChan, baseJob, "purgeBatch", &id, &tempDBs, &idMutex, &wg)
 	}
 
-	db.Close()
+	wg.Wait()
 
-	errorcount, _ := w.running.Load("errors")
-	if errorcount.(int) != 0 {
-		return errors.New("erreurs constatées, consultez les journaux")
+	db, _ := mgo.Dial(viper.GetString("DB_DIAL"))
+	db.SetSocketTimeout(720000 * time.Second)
+
+	for _, tempDB := range tempDBs {
+		pipeline := []bson.M{{
+			"$merge": bson.M{"into": bson.M{"coll": "RawData", "db": viper.GetString("DB")}}}}
+		pipe := db.DB(tempDB).C("TemporaryCollection").Pipe(pipeline)
+
+		err = pipe.AllowDiskUse().All(&[]interface{}{})
+		if err != nil {
+			fmt.Println("quelque chose vient de se casser: " + err.Error())
+			return err
+		} else {
```

Comment:
Codacy found an issue: [if block ends with a return statement, so drop this else and outdent its block](https://app.codacy.com/gh/signaux-faibles/opensignauxfaibles/pullRequest?prid=6358159)